src/Middleware/InitializeTenancy.php

```diff
@@ -5,9 +5,15 @@
 namespace Stancl\Tenancy\Middleware;
 
 use Closure;
+use Stancl\Tenancy\Exceptions\TenantCouldNotBeIdentifiedException;
 
 class InitializeTenancy
 {
+    /**
+     * @var \Closure
+     */
+    protected $onFail;
+
     public function __construct(Closure $onFail = null)
```

Comment:
```suggestion
     public function __construct(callable $onFail = null)
```

I think this should be callable, not just `Closure`. Thoughts @antonkomarev?

Reply:
It seems reasonable to use `callable` type here.